apollo-compiler/src/main/kotlin/com/apollostack/compiler/GraphqlCompiler.kt

```diff
@@ -34,12 +34,12 @@ open class GraphqlCompiler {
 
   private fun String.formatPackageName(): String {
     val parts = split(File.separatorChar)
-    val srcFolderIndex = parts.indexOfFirst { it == "src" }
-    val graphqlFolderIndex = parts.indexOfFirst { it == "graphql" }
-    if (graphqlFolderIndex - srcFolderIndex != 2) {
-      throw IllegalArgumentException("Files must be organized like src/main/graphql/...")
-    }
 
-    return parts.subList(graphqlFolderIndex + 1, parts.size).dropLast(1).joinToString(".")
+    for (i in 2..parts.size) {
+      if (parts[i - 2] == "src" && parts[i] == "graphql") {
+        return parts.subList(i + 1, parts.size).dropLast(1).joinToString(".")
```

Comment:
oh I worded that a bit poorly.

Let me reword with an example:

For the following case: `/Users/name/Desktop/src/github/apollo-client/src/main/graphql`, and using the old logic:

`parts.indexOfFirst { it == "src" }` would return 3 and `parts.indexOfFirst { it == "graphql" }` would return 8. This fails the check so an exception is thrown.

With the new change, it would break out of the loop iff it finds the correct `src` string. ie. The one that's at a distance 2 from the `graphql` string.